### PR TITLE
HTC-774/messaging connect backend

### PR DESCRIPTION
**Pull request by QinAlex**
# [#774 ](https://github.com/rachellegelden/Home-Together-Canada/issues/774)

## Summary
Connect backend to frontend.

## Relevant Motivation & Context
We need to connect backend to frontend. 
Note: 
- It is not a real time messaging system, you have to go to another nav then come back to see the new messages.
- After send a message, the text area is not empty, will have the value message content that you just send.

## Testing Instructions
1. Create 2 users (userA and userB) in post postman 
2. In postman login as 1 userA , and create some messages that send to userB
3. In post man logout as userA, login as userB, and create some messages that send to userA
4. login in as userA in http://localhost:3002/, then go to messaging
5. Clicks on the conversation card to go in to the conversation detail between 2 users
6. View and check the output
7. Create a message in textarea, and hit send button
8. Go to main page or other nav then come back to messaing, the message you just sent should display in both "converation card" and "conversation detail"
9. Logout as userA in http://localhost:3002/, then login as userB,
10. delete userB account, re-login as userA
11. Go to messaging and trying to send the message to userB again it will show the alert of that user account is deleted.

## Developer checklist prior to opening this pull request:
- [x] PR merges to the applicable branch (develop or feature branch)
- [x] Commits adhere to GitHub compliance (Issue #)
- [x] Comments for non-trivial changes
- [x] No build or runtime warnings or errors introduced
- [x] If CSS changes were introduced, change viewed in Chrome, Firefox and IE
- [x] Unit test coverage for features
- [x] Unit tests pass
- [x] Automation tests pass 

## Reviewer
- [x] Checkout and launch this branch locally
- [x] Review code structure
- [x] Review test coverage
- [x] If CSS changes were introduced, change viewed were viewed in Chrome, Firefox and IE

### CODE REVIEW

**Comment by Lightbrite88** [comment]
Thanks for leaving the notes of the limitations

**Comment by Lightbrite88** [comment]
reminder @rachellegelden when you look at this you will need to delete database and recreate it so tables are correct - some changes were to the model

File: client/src/accountSummary/member/messaging/SendMessage.js [hunk 1]
```diff
@@ -47,7 +47,7 @@ function SendMessage(props) {
                     if (data.sent) {
                         alert("Message sent successfully");
                     } else {
-                        alert("Message not sent successfully")
+                        alert("Sorry, the person you are trying to reach has deleted his/her account.")
```
Comment (Lightbrite88) [inline client/src/accountSummary/member/messaging/SendMessage.js:50]: Can we change this to neuttral "their" account instead of "his/her"?

Comment (Lightbrite88) [reply client/src/accountSummary/member/messaging/SendMessage.js:50]: please

**Review by Lightbrite88 — CHANGES_REQUESTED** [review]
This looks great alex - one small trivial change

**Review by Lightbrite88 — APPROVED** [review]
Looks good. Thanks alex

File: server/controllers/abstractUserController.js [hunk 2]
```diff
@@ -47,6 +47,14 @@ const findUsersByUsernames = (usernames) =>
...
     });
 
+// added
+const findUserByUserId = uid =>
+    AbstractUser.findOne({
+        where: {
+            uid: uid
+        }
+    })
+
```
Comment (rachellegelden) [inline server/controllers/abstractUserController.js:57]: This function does the exact same thing as `findAbstractUser`. Please use that function and get rid of this function

File: server/models/message.js [hunk 3]
```diff
@@ -11,10 +11,18 @@ module.exports = (DataTypes, sequelize) => {
...
+            allowNull: false,
+        },
         receiverId: {
             type: DataTypes.INTEGER,
             allowNull: false,
         },
+        receiverName: {
+            type: DataTypes.STRING,
+            allowNull: false,
+        },
```
Comment (rachellegelden) [inline server/models/message.js:25]: These fields are duplicates of the `username` field in `abstractUser` model. If you need to access these fields, do a join on the `abstractUser` model. Please get rid of them here

**Review by rachellegelden — CHANGES_REQUESTED** [review]
Just a couple changes

File: server/controllers/abstractUserController.js (hunk 2 above)
Comment (QinAlex) [reply server/controllers/abstractUserController.js:57]: Addressed.

File: server/models/message.js (hunk 3 above)
Comment (Lightbrite88) [reply server/models/message.js:25]: I can't speak to recieverName @QinAlex can elaborate I am sure, but senderName has to be stored with the message else we lose access to it if that user is deleted. I.e Receiver keeps the message, but it looks like it came from from no one. @rachellegelden

File: server/models/message.js [hunk 4]
```diff
@@ -11,10 +11,18 @@ module.exports = (DataTypes, sequelize) => {
...
         },
+        senderName: {
+            type: DataTypes.STRING,
+            allowNull: false,
+        },
         receiverId: {
             type: DataTypes.INTEGER,
             allowNull: false,
         },
+        receiverName: {
```
Comment (Lightbrite88) [inline server/models/message.js:22]: Please change receiverName to receiverUsername

**Review by Lightbrite88 — CHANGES_REQUESTED** [review]
Please make sure both the conversation card and the fullconversation have some message stating that a user is deleted.

File: client/src/accountSummary/member/messaging/SendMessage.js (hunk 1 above)
Comment (Lightbrite88) [reply client/src/accountSummary/member/messaging/SendMessage.js]: addressed

**Pull request closed without merging**